### PR TITLE
Events/Analytics/Exporter

### PR DESCRIPTION
**Pull request by niwinz**

### CODE REVIEW

File: backend/src/app/util/logging.clj [hunk 1]
```diff
@@ -60,17 +60,21 @@
...
        (if (enabled? ~logger-sym ~level-sym)
          ~(if async
-            `(send-off logging-agent (fn [_#] (write-log! ~logger-sym ~level-sym ~cause (build-map-message ~props))))
-            `(write-log! ~logger-sym ~level-sym ~cause (build-map-message ~props)))))))
+            `(send-off logging-agent
+                       (fn [_#]
+                         (let [message# (or ~raw (build-map-message ~props))]
+                           (write-log! ~logger-sym ~level-sym ~cause message#))))
+            `(let [message# (or ~raw (build-map-message ~props))]
+               (write-log! ~logger-sym ~level-sym ~cause message#)))))))
```
Comment (hirunatan) [inline backend/src/app/util/logging.clj:77]: I think you can move the `(let` to before the `(if async`, this way you only need to write it once

File: frontend/src/app/main/data/events.cljs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                                       (when (fn? impl-fn)
+                                         (some-> (impl-fn event)
+                                                 (update :profile-id #(or % profile-id)))))))
+                           (rx/filter :profile-id)
+                           (rx/map (fn [event]
+                                     (let [session* (or @session (dt/now))
+                                           context  (-> @context
+                                                        (d/merge (:context event))
+                                                        (assoc :session session*))]
+                                       (swap! session (constantly session*))
```
Comment (hirunatan) [inline frontend/src/app/main/data/events.cljs]: Could it be `(reset! session session*)`?

File: backend/src/app/util/logging.clj (hunk 1 above)
Comment (niwinz) [reply backend/src/app/util/logging.clj:77]: No, this makes the message building synchronous, and the purpose of async is that message building is offlodead to an other thread.

File: frontend/src/app/main/data/events.cljs (hunk 2 above)
Comment (niwinz) [reply frontend/src/app/main/data/events.cljs]: you are right, the swap! here is unnecesary. I'll change it.

File: backend/src/app/util/logging.clj (hunk 1 above)
Comment (hirunatan) [reply backend/src/app/util/logging.clj:77]: Okkk i didn't notice that  `(fn [_#])`

**Pull request merged**